src/controller/CHIPDeviceController.cpp

```diff
@@ -103,6 +103,7 @@ constexpr uint32_t kSessionEstablishmentTimeout = 30 * kMillisecondPerSecond;
 constexpr uint32_t kMaxCHIPOpCertLength = 1024;
 constexpr uint32_t kMaxCHIPCSRLength    = 1024;
 constexpr uint32_t kOpCSRNonceLength    = 32;
+constexpr uint32_t kMaxPayloadSize      = 1280;
```

Comment:
This is too large. 1280 is the max packet size. The ping payload has to fix after all message layer headers

Reply:
VerifyOrExit(size <= kMaxPayloadSize, err = CHIP_ERROR_MESSAGE_TOO_LONG);
payloadBuf = MessagePacketBuffer::NewWithData(requestData, size);

This size is used to allocate CHIP message buffer, it contains all message layer headers

Reply:
`MessagePacketBuffer::NewWithData` will take the passed-in size, add the length of a MIC to that, pass that to PacketBuffer to allocate a buffer with that much payload.  Which will then add the default header size on to that and allocate _that_.

So if `size == kMaxPayloadSize` the above code will try to allocate a PacketBuffer with storage for `1280 + 16 + SystemPacketBuffer::kDefaultHeaderReserve` bytes, right?  Is that expected?

I suspect the right value for `kMAxPayloadSize` is more like `SystemPacketBuffer::kMaxSize - MessagePacketBuffer::kMaxFooterSize` or something.

Or just leave out the size check and just deal with the `NewWithData` failing due to it being too big?

Reply:
I prefer leave it to the size check within NewWithData